x-pack/plugins/observability_ai_assistant/server/service/client/index.ts

```diff
@@ -345,6 +347,67 @@ export class ObservabilityAIAssistantClient {
     });
   };
 
+  get_dataset_info = async (
+    index: string
+  ): Promise<{
+    indices: string[];
+    fields: Array<{ name: string; description: string; type: string }>;
+  }> => {
+    // if empty dataview, get list of all dataviews
+
+    let indices: string[] = [];
+
+    try {
+      const body = await this.dependencies.esClient.indices.resolveIndex({
+        name: index === '' ? '*' : index,
+        expand_wildcards: 'open',
+      });
+      indices = body.indices.map((i) => i.name);
+    } catch (e) {
+      indices = [];
+    }
+
+    if (index === '') {
+      return {
+        indices,
+        fields: [],
+      };
+    }
+
+    if (indices.length === 0) {
+      try {
+        const body = await this.dependencies.esClient.indices.resolveIndex({
+          name: '*',
+          expand_wildcards: 'open',
+        });
+        indices = body.indices.map((i) => i.name);
+      } catch (e) {
+        indices = [];
+      }
+
+      return {
+        indices,
+        fields: [],
+      };
+    }
```

Comment:
Does this mean if no indices are found for the given `index`, it will return all indices? Doesn't this confuse the LLM?

Reply:
sometimes LLM will still assume user entered correct index name and will try to use it. in case it fails it gets full index list back and it seems to figure out from it that the index name provided was not correct and it tries to find the right one based on the list provided.

Reply:
i actually changed this, so if no indices match then empty index array and empty field array qre returned